### DISCUSSION TITLE
Security Vulnerability - Students can claim to be other students

### DISCUSSION THREAD

**Discussion by cstest1** [community/community#138916] [GitHub Education]
### Hi everyone,

Vulnerability: Students can claim to be other students.

STR:
On Teacher Account:
Set up a github classroom and add students complete with email addresses.
Make an assignment.
Send students the link to the assignment.

On Student Account:
Create a github account using the email address that the teacher used when adding their roster
Click on the assignment link and log in.

Expected result: Github associates your github classroom account with your email address that you're logged into to the student record with the same email address, or sends you and the teacher an error message if you're not in the list.

Actual result: "Choose which student you are", which allows you to choose to be someone else, with a different email address.

![GitHub Fail](https://github.com/user-attachments/assets/cdf471f8-74c3-47ef-99ea-e259fdf112fd)
d

**Comment by Ujoa** [comment]
Yeah. theres no feature to upload a mapping of students emails/id to names in classrooms.

This is a known limitation and sucks. I constantly have students emailing me during the first week or two saying they mapped to the wrong person on accident

**Reply by g-simmons** [reply]
Agree. 

Not just a security problem but a logistics headache.

Students report that it is easy to accidentally click someone else's name.
